temporalio/converter.py

```diff
@@ -732,16 +732,17 @@ def _type_hints_from_func(
     # the func. This mimics inspect logic inside Python.
     if (
         not inspect.isfunction(func)
-        and not isinstance(func, type)
```

Comment:
I recall there being some concern that this will catch types we don't want to here. If that is the case hopefully the tests catch it then.